File: code/python/playground/playground-fastapi/quick-startup/apps/example/controller/request.py
# coding: UTF-8
"""
@software: PyCharm
@author: Lionel Johnson
@contact: https://fairy.host
@organization: https://github.com/FairylandFuture
@datetime: 2025-09-13 15:06:06 UTC+08:00
"""

import typing as t
from fastapi import Request

from utils.http.response import Response


class RequestController:

    async def request_info(self, request: Request) -> Response:
        print(f"请求URL: {request.url}")
        print(f"请求方法: {request.method}")
        print(f"请求头: {request.headers}")
        print(f"请求客户端: {request.client}")
        print(f"请求cookies: {request.cookies}")
        print(f"请求路径参数: {request.path_params}")
        print(f"请求查询参数: {request.query_params}")

        data = {
            "url": request.url,
            "method": request.method,
            "headers": request.headers,
            "client": request.client,
        }

        return Response(data=data)

File: code/javascript/mock/request.js
/****************************************************
 * @software: PyCharm
 * @author: Lionel Johnson
 * @contact: https://fairy.host
 * @organization: https://github.com/FairylandFuture
 * @datetime: 2025-09-14 11:51:30 UTC+08:00
 ***************************************************/

export async function fetchTest() {
    return new Promise((resolve, reject) => {
        $.ajax({
            url: 'http://127.0.0.1:51005/example/test',
            method: 'GET',
            dataType: 'text',
            timeout: 10000,
            success: function (data, textStatus, jqXHR) {
                resolve({
                    data,
                    textStatus,
                    status: jqXHR.status,
                    headers: jqXHR.getAllResponseHeaders()
                });
            },
            error: function (jqXHR, textStatus, errorThrown) {
                const err = new Error(errorThrown || textStatus || 'AJAX error');
                err.status = jqXHR && jqXHR.status;
                err.responseText = jqXHR && jqXHR.responseText;
                reject(err);
            }
        });
    });
}

export default fetchTest;